src/dart/parse.rs

```diff
@@ -65,6 +78,16 @@ impl<'a> Parser<'a> {
         }
     }
 
+    fn peek(&mut self) -> Option<&'a Token> {
+        while let Some(token) = self.next_token() {
+            self.buffer.push_back(token);
+            if token.is_whitespace() {
```

Comment:
You accidentally removed the negation here - we want non-whitespace tokens, not whitespace tokens 😆.